typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -677,4 +689,23 @@ export abstract class HyperlaneDeployer<
 
     return receipts.filter((x) => !!x) as ethers.ContractReceipt[];
   }
+
+  protected async resolveInterchainAccountAsOwner(
+    chain: ChainName,
+    owner: Owner,
+  ): Promise<Address> {
+    if (typeof owner === 'string') {
+      return owner;
+    } else {
+      const routerAddress = this.cachedAddresses[chain].interchainAccountRouter;
```

Comment:
afaict this means ICAs would show up in warp route deploy artifacts

Reply:
Ya that's problem.

Do you mean `deployedIsms` in the deployer? I felt this would have been a more invasive change to the basedeployer? We should discuss this tmrw morning.

Reply:
no I mean similar to the `ismFactory` on the base deployer, having a `icaApp` on the base deployer as a deploy option

Reply:
ya that's what I meant

Reply:
fixed